cpp/dolfinx/fem/FunctionSpace.h

```diff
@@ -62,7 +61,11 @@ class FunctionSpace
   /// Move assignment operator
   FunctionSpace& operator=(FunctionSpace&& V) = default;
 
-  /// Extract subspace for component
+  ///  @brief Extracts a subspace for a specific component. 
+  ///
+  /// @note For performance reasons, it's advisable to store and reuse the
+  /// returned subspace instead of calling this function repeatedly for the same
+  /// component.
   /// @param[in] component The subspace component
   /// @return The subspace
   std::shared_ptr<FunctionSpace<T>> sub(const std::vector<int>& component) const
```

Comment:
I think it would be clearer if the return type was `FunctionSpace<T>` (not a shared_ptr) and we remove the cache.

Reply:
I've updated the return type.